receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,337 @@
+		for _, hostname := range hostnames {
+			r.Cluster = &cluster
+			r.Hostname = hostname
+			s.sendLogs(r, "mongodb.gz")
+			s.sendLogs(r, "mongos.gz")
+
+			if cfgProjects[r.Project.Name].EnableAuditLogs {
+				s.sendAuditLogs(r, "mongodb-audit-log.gz")
+				s.sendAuditLogs(r, "mongos-audit-log.gz")
+			}
+		}
+	}
+
+	return nil
+
+}
+
+func filterClusters(clusters []mongodbatlas.Cluster, keys map[string]struct{}, include bool) ([]mongodbatlas.Cluster, error) {
+	var filtered []mongodbatlas.Cluster
+	for _, cluster := range clusters {
+		if _, ok := keys[cluster.Name]; (!ok && !include) || (ok && include) {
+			filtered = append(filtered, cluster)
+		}
+	}
+	return filtered, nil
+}
+
+func createStringSet(in []string) map[string]struct{} {
+	list := map[string]struct{}{}
+	for i := range in {
+		list[in[i]] = struct{}{}
+	}
+
+	return list
+}
+
+func (s *receiver) getHostLogs(groupID, hostname, logName string) ([]model.LogEntry, error) {
+	// Get gzip bytes buffer from API
+	buf, err := s.client.GetLogs(context.Background(), groupID, hostname, logName)
+	if err != nil {
+		return nil, err
+	}
+	// Pass this into a gzip reader for decoding
+	reader, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Logs are in JSON format so create a JSON decoder to process them
+	dec := json.NewDecoder(reader)
+
+	entries := make([]model.LogEntry, 0)
+	for {
+		var entry model.LogEntry
+		err := dec.Decode(&entry)
+		if errors.Is(err, io.EOF) {
+			return entries, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, entry)
+	}
+}
+
+func (s *receiver) getHostAuditLogs(groupID, hostname, logName string) ([]model.AuditLog, error) {
+	// Get gzip bytes buffer from API
+	buf, err := s.client.GetLogs(context.Background(), groupID, hostname, logName)
+	if err != nil {
+		return nil, err
+	}
+	// Pass this into a gzip reader for decoding
+	reader, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Logs are in JSON format so create a JSON decoder to process them
+	dec := json.NewDecoder(reader)
+
+	entries := make([]model.AuditLog, 0)
+	for {
+		var entry model.AuditLog
+		err := dec.Decode(&entry)
+		if errors.Is(err, io.EOF) {
+			return entries, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, entry)
+	}
+}
+
+func (s *receiver) sendLogs(r resourceInfo, logName string) {
+	logs, err := s.getHostLogs(r.Project.ID, r.Hostname, logName)
+	if err != nil && err != io.EOF {
+		s.log.Warn("Failed to retreive logs", zap.Error(err))
+	}
+
+	for _, log := range logs {
+		r.LogName = logName
+		plog := mongodbEventToLogData(s.log, &log, r)
+		s.consumer.ConsumeLogs(context.Background(), plog)
+	}
+}
+
+func (s *receiver) sendAuditLogs(r resourceInfo, logName string) {
+	logs, err := s.getHostAuditLogs(r.Project.ID, r.Hostname, logName)
+	if err != nil && err != io.EOF {
+		s.log.Warn("Failed to retreive logs", zap.Error(err))
```

Comment:
should these messages on both functions specify the type of logs that were failed to be retrieved?